lib/gringotts/gateways/pin_payments.ex

```diff
@@ -126,20 +126,24 @@ defmodule Gringotts.Gateways.PinPayments do
   def authorize(amount, %CreditCard{} = card, opts) do
     {currency, value, _} = Money.to_integer(amount)
 
-    card_token =
+    card_token_response =
       :post
       |> commit("cards", card_for_token(card, opts) ++ Keyword.delete(opts, :address))
       |> extract_card_token
 
-    params =
+      case card_token_response do
+         {:error, error} -> {:error, Response.error(code: error.id, message: "HTTPoison says '#{error.reason}'")}
```

Comment:
What if the response is *not 200*? You'd still get an `:ok` tuple, but it will not have any `token` in it.